ddtrace/contrib/mako/patch.py

```diff
@@ -40,7 +41,16 @@ def _wrap_render(wrapped, instance, args, kwargs):
     if not pin or not pin.enabled():
         return wrapped(*args, **kwargs)
 
-    template_name = instance.filename or DEFAULT_TEMPLATE_NAME
+    # Determine the resource and `mako.template_name` tag value
+    template_name = None
+    # DefTemplate is a wrapper around a callable from another template, it does not have a filename
+    # https://github.com/sqlalchemy/mako/blob/c2c690ac9add584f2216dc655cdf8215b24ef03c/mako/template.py#L603-L622
+    if isinstance(instance, DefTemplate) and hasattr(instance, "callable_"):
+        template_name = func_name(instance.callable_)
+    else:
+        template_name = getattr(instance, "filename", None)
```

Comment:
```suggestion
        template_name = getattr(instance, "filename", DEFAULT_TEMPLATE_NAME)
```

Reply:
The problem is if it exists but it is `None`, we still want to fallback to the `DEFAULT_TEMPLATE_NAME`, otherwise the tag won't get set/exist.

Unless we'd prefer a different value to be used for "nothing was set"

Reply:
Not sure I have a very strong opinion here, it is mostly to guard against the chance of `template_name` ever being `None`

Reply:
Ah gotcha. Okay that second check makes sense then!

Reply:
Maybe worth using a sentinel here so we can know the difference? Might be misleading to report the default when the filename is actually there but `None`

Reply:
I am open to suggestion, I couldn't figure out a good value... omitting the tag entirely could be an option, but we now have precedence for a default of `<memory>` for anyone using it today :/

Reply:
Fair enough, let's leave it as-is then :+1: